### DISCUSSION TITLE
Formatting code

### DISCUSSION THREAD

**Discussion by clavay** [pyscada/PyScada#81] [Ideas]
I think we should format code using some tools for any new PR in any plugin.

- [x] For python we already use [black](https://black.readthedocs.io/en/stable/index.html).
- [ ] For django template, css and js we can use : [djhtml](https://github.com/rtts/djhtml).
- [ ] We should add this requisite in the readme.
- [ ] We can use [pre-commit](https://pre-commit.com) for identifying simple issues before submission to code review and code formatting automation.

**Comment by trombastic** [comment]
i did some testing with `pre-commit` on the pyscada-ems plugin (not jet pushed to github), looks good to me. I will do some more tests on the pyscada repo. I Used the guidelines from django and used the config files from django as template. We need to add a `pyproject.toml`  file for this to work.

**Reply by clavay** [reply]
Ok. If you put this file in a branch, I can test it and validate the pre-commit use.

**Reply by trombastic** [reply]
i pushed my recent changes to the pyscada-ems plugin which included the pre-commit stuff

**Reply by clavay** [reply]
Ok, so we should use black, flake8, isort, eslint, and black for python code in docs.

Why didn't you add djhtml in the pre-commit config ?
